modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/RefreshTokenHandler.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * 
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.mdm.agent.proxy;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
+import android.os.AsyncTask;
+import android.util.Log;
+import org.apache.commons.codec.binary.Base64;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Getting new access token and refresh token after access token expiration
+ */
+public class RefreshTokenHandler {
+	private static final String TAG = "RefreshTokenHandler";
+	private static Token token;
+
+	public RefreshTokenHandler(Token token) {
+		RefreshTokenHandler.token = token;
+	}
+
+	public void obtainNewAccessToken() throws InterruptedException, ExecutionException,
+	                                          TimeoutException {
+		new NetworkCallTask().execute();
```

Comment:
Rather than creating just an ad-hoc instance of NetworkCallTask, is there any possibility for us to use a TaskPool (Implemented using commons-pool, etc) and control instantiating the aforesaid objects?